src/main/java/seedu/address/model/drink/UniqueDrinkList.java

```diff
@@ -0,0 +1,135 @@
+package seedu.address.model.drink;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Iterator;
+import java.util.List;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import seedu.address.model.drink.exceptions.DrinkNotFoundException;
+import seedu.address.model.drink.exceptions.DuplicateDrinkException;
+
+/**
+ * A list of drinks that enforces uniqueness between its elements and does not allow nulls.
+ * A drink is considered unique by comparing using {@code Drink#isSameDrink(Drink)}. As such, adding and updating of
+ * drinks uses Drink#isSameDrink(Drink) for equality so as to ensure that the drink being added or updated is
+ * unique in terms of identity in the UniqueDrinkList. However, the removal of a drink uses Drink#equals(Object) so
+ * as to ensure that the drink with exactly the same fields will be removed.
+ * <p>
+ * Supports a minimal set of list operations.
+ *
+ * @see Drink#isSameDrink(Drink)
+ */
+public class UniqueDrinkList implements Iterable<Drink> {
+
+    private final ObservableList<Drink> internalList = FXCollections.observableArrayList();
+
+    /**
+     * Returns true if the list contains an equivalent drink as the given argument.
+     */
+    public boolean contains(Drink toCheck) {
+        requireNonNull(toCheck);
+        return internalList.stream().anyMatch(toCheck::isSameDrink);
+    }
+
+    /**
+     * Adds a drink to the list.
+     * The drink must not already exist in the list.
+     */
+    public void add(Drink toAdd) {
+        requireNonNull(toAdd);
+        if (contains(toAdd)) {
+            throw new DuplicateDrinkException();
+        }
+        internalList.add(toAdd);
+    }
+
+    /**
+     * Replaces the drink {@code target} in the list with {@code editedDrink}.
+     * {@code target} must exist in the list.
+     * The drink identity of {@code editedDrink} must not be the same as another existing drink in the list.
+     */
+    public void setDrink(Drink target, Drink editedDrink) {
```

Comment:
Are we support this?? ( I assuming this means that change the name of the drink only)
as I prefer delete and create again rather than let them change...
As once u have a batch inside drink name, it is very weird to change...( and can we change it??)

Reply:
based on what we've discussed on Monday, we will be be _editing_ the attributes (including name) of the drink, instead of deleting and adding. This will entail finding the drink in question (by name) and having the edited name as one of the parameters. The Name object will then be replaced.